rtlib/src/Data.cpp

```diff
@@ -0,0 +1,52 @@
+//
+// Created by Anthony Tamasi on 2019-10-31.
+//
+
+#include "rt/Data.hpp"
+
+#include <itkOpenCVImageBridge.h>
+#include <nlohmann/json.hpp>
+
+#include "rt/Envi.hpp"
+#include "rt/Image.hpp"
+#include "rt/Mesh.hpp"
+
+
+namespace fs = boost::filesystem;
+using json = nlohmann::json;
+using namespace rt;
+
+// IO
+using OCVBridge = itk::OpenCVImageBridge;
+
+enum class Type { Image, Envi, Mesh };
+
+Data::Pointer Data::Load(const fs::path& path)
+{
+    fs::path metadata_file;
+
+    //Find this file path's metadata file
+    for (fs::directory_entry& entry : fs::directory_iterator(path)){
+        if(entry.path().stem() == "metadata" && entry.path().extension() != "json") {
+            metadata_file = entry.path();
+        }
+    }
+
+    //Read "spatial object type" of the metadata file into JSON object
+    std::ifstream input_stream(metadata_file.string());
+    json j;
+    input_stream >> j;
+
+    Type metadata_type = j["type"];
```

Comment:
You need to give these types explicit values if you're going to serialize them:
```c++
enum class Type { Image = 0, Envi, Mesh };

Type metadata_type{Type::Envi};
j["type"] = metadata_type; // serialize to json. Nlohmann does so as int
metadata_type = static_cast<Type>(j["type"]); // deserialize from json to Type
```

This can be better controlled here: https://github.com/nlohmann/json#specializing-enum-conversion